### PR TITLE
Add last two validateAddress tests

### PR DESCRIPTION
**Pull request by DevMandy**
Added tests for DX-1394 and DX-1387 for `validateAddress`.

I then realized that I did not have a check for `phone` in my `assertAddressEquals` method and had to add it.
I then had to add a `phone` property to the existing tests.

### CODE REVIEW

File: test/specs/validate-address.spec.js [hunk 1]
```diff
@@ -694,9 +704,101 @@ describe("validateAddress()", () => {
...
+    // The isValid flag is true
+    expect(isValid).to.be.a("boolean").and.to.equal(true);
+
+    // The normalized address is populated and matches the expected normalized address
+    assertAddressEquals(normalizedAddress, expectedNormalizedAddress);
+
+    // The isResidential flag on the normalized address is true
+    expect(normalizedAddress.isResidential)
+      .to.be.a("boolean")
+      .and.to.equal(false);
```
Comment (JamesMessinger) [inline test/specs/validate-address.spec.js:745]: Why test for this separately? Why not just include it in the `expectedNormalizedAddress` object?

File: test/specs/validate-address.spec.js [hunk 2]
```diff
@@ -694,9 +704,101 @@ describe("validateAddress()", () => {
...
+    // The isValid flag is true
+    expect(isValid).to.be.a("boolean").and.to.equal(true);
+
+    // The normalized address is populated and matches the expected normalized address
+    assertAddressEquals(normalizedAddress, expectedNormalizedAddress);
+
+    // The isResidential flag on the normalized address is true
+    expect(normalizedAddress.isResidential)
+      .to.be.a("boolean")
+      .and.to.equal(true);
```
Comment (JamesMessinger) [inline test/specs/validate-address.spec.js:789]: Why test for this separately? Why not just include it in the `expectedNormalizedAddress` object?

**Review by JamesMessinger — COMMENTED** [review]
PR looks great, Mandy.  I'm just curious why `isResidential` isn't just tested as part of the `assertAddressEquals()` function

File: test/specs/validate-address.spec.js (hunk 2 above)
Comment (DevMandy) [reply test/specs/validate-address.spec.js:789]: Just to be explicit. I can definitely add it there as well.

File: test/specs/validate-address.spec.js (hunk 1 above)
Comment (DevMandy) [reply test/specs/validate-address.spec.js:745]: Just to be explicit. I can definitely add it there as well. If there is a specific criteria in the test, I have asserted outside of a the method just to make it explicit, but i can change that.

Comment (JamesMessinger) [reply test/specs/validate-address.spec.js:745]: The criteria for these tests don't have anything to do with the `isResidential` field though.  These tests are about the `name`, `company`, and `phone` fields.

**Pull request closed without merging**